### PR TITLE
[OSF-7994] Fix loss of binding for contributor adder in registration editor.

### PR DESCRIPTION
**Pull request by TomBaxter**
Cache contributor adder to avoid loss of binding when navigating between
pages of registration editor.
Added some lodash cause, why not more lodash.

## Purpose

Fix loss of binding for contributor adder in registration editor.

## Changes

Update packages.json Added some lodash
Update website/static/js/registrationUtils.js Added facility to cache some extensions instead of recreating them. For instance osf-author-import loses it's binding when recreated.

## Side effects
None

## Ticket

[OSF-7994](https://openscience.atlassian.net/browse/OSF-7994)

### CODE REVIEW

File: website/static/js/registrationUtils.js [hunk 1]
```diff
@@ -37,6 +40,9 @@ $.each(VALIDATORS, function(key, value) {
     VALIDATOR_LOOKUP[value.message] = value;
 });
 
+// Extentions that lose data bind if re-created.
```
Comment (alexschiller) [inline website/static/js/registrationUtils.js]: Very Minor: typo, `Extensions`

File: website/static/js/registrationUtils.js [hunk 2]
```diff
@@ -1005,7 +1011,15 @@ RegistrationEditor.prototype.context = function(data, $root, preview) {
     });
 
     if (this.extensions[data.type]) {
-        return new this.extensions[data.type](data, $root.pk, preview);
+        // osf-author-import loses binding when re-created. Hence, this kludge.
+        if (lodashIncludes(CACHED_EXTENSIONS, data.type)) {
+            if (!lodashHas(this, ['extCache', data.type, data.id])) {
+                lodashSet(this, ['extCache', data.type, data.id],  new this.extensions[data.type](data, $root.pk, preview));
```
Comment (alexschiller) [inline website/static/js/registrationUtils.js]: extra space before `new`

File: website/static/js/registrationUtils.js (hunk 1 above)
Comment (TomBaxter) [inline website/static/js/registrationUtils.js]: Fixed

File: website/static/js/registrationUtils.js (hunk 2 above)
Comment (TomBaxter) [inline website/static/js/registrationUtils.js]: Fixed

**Comment by TomBaxter** [comment]
@alexschiller Did you want to look at this any further? Or should I throw it back into the the general CR pool?

**Comment by alexschiller** [comment]
@TomBaxter It can go back in the queue, no other comments 👍

**Pull request merged**